### DISCUSSION TITLE
Age and Growth

### DISCUSSION THREAD

**Discussion by sabrinabeyer** [pfmc-assessments/shortspine_thornyhead_2023#5] [General]
Hi Age and Growth team!

I am starting a discussion channel for us. Updates to come! @shipmadison @JaneSullivan-NOAA @MatthVeron @AndreaNOdell

**Comment by AndreaNOdell** [comment]
Hi Growth Team! Have we been meeting regularly? I realize I'm a bit out of the loop and want to check in to see if i've been missing anything. I last attended our meeting with Donna

**Reply by MatthVeron** [reply]
Thanks @AndreaNOdell for jumping up on this, I was about sending an email today. It might be great to meet tomorrow if you guys are available to discuss what has been done. I think @sabrinabeyer has done a bunch of things (thanks again for this) so it might be cool if we can figure out a plan for the next steps.

**Reply by AndreaNOdell** [reply]
Great - I am available anytime between 11:30am and 4pm!

**Reply by sabrinabeyer** [reply]
Hi, I am available to meet tomorrow!

Donna sent quite a few files. I think most are on the g-drive in 2023 SST > assessment inputs > parameters > SST Age data. Caution that data are duplicated across files. I used the information Donna provides to create two "formatted" files for the Kline 1996 and Butler et al. 1995 data that run with the associated R code in the folder. I will need help uploading the R code to Github. Feel free to explore! 

An updated graph and questions are on the g-slides.

**Reply by MatthVeron** [reply]
Thanks @sabrinabeyer for all of this. I can help you to push stuff on guthub if you want ! 
Awesome, let's plan a meeting at 1:30pm  if that works for you guys !

**Comment by sabrinabeyer** [comment]
Hi! I will work with Matthieu today to get R code and data files on Github.

One issue to solve...Andrea was right and there are still duplicated fish in the Butler et al dataset (Kline 1996 dataset is ok). The issue is that in their later surveys, they collected both otoliths from a single fish and there may be multiple reads. So we should come up with some code to exclude (or combine) an age estimate from both otoliths of a single fish (does that make sense)?

**Reply by JaneSullivan-NOAA** [reply]
Yes it makes sense! I can take a look at it later today once you all have the data and code up on github if you want.

**Reply by sabrinabeyer** [reply]
Thanks! 

Matthieu helped me to push code to Github (although I will edit what I pushed to make it a bit more clear). He also found a way to identify the "duplicates" (i.e. a fish that had two otoliths pulled and multiple age reads for each otolith).

A question for the group is how would we like to use that information? For example, in the Butler et al 1995 analysis (and what is currently in our code) is that we take the mean age for readers AR and JB and use that as the "Age_for_analysis". Should we take use the mean age for all readers of all otoliths to use for the age of a single fish? Lot's of different ways to try this....

**Comment by sabrinabeyer** [comment]
Next steps with age data....

...if you're feeling eager and ambitious, jump on in and give these a try! Let us know if you start working on something so we don't duplicate efforts??

1. Figure out what to do with "duplicates" in Butler age data (i.e. multiple reads of both otoliths for a single fish)
         -Does this solve any of the confusion in differences among growth curves?

2. There is some confusion about what was used for the parameters in the 2013 assessment...it is possible that they used t0 (VBGF) for age 1 (Schnute)?? Try plotting the curve from the 2013 assessment with the VBGF parameterization (not the Schnute re-parameterization)---does this match the Kline 1996 data?

3. Pool all data from both studies and fit a curve just as a check to see if that re-creates the 2013 assessment parameters (not recommended to use in this assessment, but as a check to see if this may have been done in the past)

**Reply by sabrinabeyer** [reply]
4. Make a better looking ggplot!

**Reply by JaneSullivan-NOAA** [reply]
This list looks reasonable to me @sabrinabeyer! I'll try to work through some of it this afternoon and will provide an update here at the end of the day

**Reply by JaneSullivan-NOAA** [reply]
I'm making some progress but am having an issue with '2rD' (what DK told us is the JB's second read on the otolith). In all but two cases, the second read is smaller than the first. On average, the second read is ~65% less than the first read and in one extreme, it's 200% less than the first read. That makes no sense at all to me... I do not think this is the accurate column definition for '2rD' and I don't think we should use it.

**Reply by sabrinabeyer** [reply]
I have no problem not using the 2rD read.

I also added three pdfs from Donna to Github in the age folder:
-one "describes" the Butler et al dataset sent from JB to DK (this is where I interpreted 2rD as a second read....but it does seem suspicious that all 2nd reads were biased younger)

-the other two are a NOAA SWFSC cruise plan (and subsequent cruise report) for the Kline 1996 fish--most of Donna's fish were collected on this NOAA cruise in 1991

**Reply by JaneSullivan-NOAA** [reply]
Thanks @sabrinabeyer! 
I think I've got the data set pretty well cleaned but unfortunately the model we've been looking at wasn't actually converged. This is a common issue with optim, especially using default settings. If you print the vbgf.optim object, you'll see $convergence of 1, which indicates that relative convergence criteria haven't been met. It looks like switching to a different optimizer (e.g. BFGS or L-BFGS-B) works without having to tweak the controls.

**Comment by JaneSullivan-NOAA** [comment]
I fit the data a few different ways (all with Schnute lognormal): 
![image](https://user-images.githubusercontent.com/68303089/221071592-8dc4856a-b6ba-4c41-bfbd-e3deae8a7d60.png)
The 2023 sex-specific L1, L2, k model was the best by AIC. I re-read the 2005 assessment. 
It does look like the Schnute values were based on von Bertalanffy values, it doesn't really seem like anything was estimated. Here's the text (note the use of the word "set"):
![image](https://user-images.githubusercontent.com/68303089/221071966-48334a90-5031-4183-be4d-ae45704bf66c.png)

**Comment by JaneSullivan-NOAA** [comment]
So this is a comparison of the 2023 sex-specific Schnute curves with the 2013 values. The shaded region would encompass the +/-10% on both the L1 and L2 parameters (last assessment only looked at 10% on L2) 
![image](https://user-images.githubusercontent.com/68303089/221258966-a8977498-d15d-4f60-a34a-3b2646bf2954.png)

**Comment by JaneSullivan-NOAA** [comment]
Code: https://github.com/JaneSullivan-NOAA/shortspine_thornyhead_2023/blob/main/R/bio/growth.R

csv file of parameters for SS data file: https://github.com/JaneSullivan-NOAA/shortspine_thornyhead_2023/blob/main/outputs/growth/growth_curve_sensitivities.csv

**Comment by JaneSullivan-NOAA** [comment]
A summary of the data cleaning process: 

**The original data set had 1,197 rows. The cleaned data set had 1,023 rows.**

1) `SIDE` column: we are not entirely sure what this means but here is the breakdown of the data:
```
   SIDE     n
  <dbl> <int>
1     0     4
2     1    86
3     2  1094
4     3     1
5    NA    12
```
The vast majority of the data are SIDE = 2, so I only analyzed fish with SIDE = 2.
2)  `DK_reader_4`:  did not age very many fish, so I did not use her age readings. 
3)  I limited the data to include samples that were aged by both AR_reader_1 and JB_reader_2
4)  I did not use the `2rD` column. We were told this was JB's second reading but that is highly unlikely given that in all but two cases, the second read is smaller than the first. On average, the second read is ~65% less than the first read, and in one extreme, it's 200% less than the first read.
5)  There is evidence of age reader bias within the data set, specifically reader 1 (AR) is more likely to age a fish older than reader
2 (JB) on average. However, once I went through data cleaning steps 1-4, I found that 95% of the remaining age reads fell within 33% absolute difference. When I trimmed the outlying 5%, there was no longer evidence of bias in the data (just high error!)
6) After this, there was only one duplicate row identified (COLL = 1310, FISH = 409, OTO = 90).
```
coll  fish   sex   oto len_mm  wt_g    r1    r2 id         
  <dbl> <dbl> <dbl> <dbl>  <dbl> <dbl> <dbl> <dbl> <chr>      
1  1310   409     2    90    198    78    10    11 1310_409_90
2  1310   409     2    90    198    78    11    11 1310_409_90
```
You can see reader 1 (AR) has two difference entries for this otolith. I removed this sample.

**Reply by JaneSullivan-NOAA** [reply]
@sabrinabeyer @AndreaNOdell @MatthVeron @shipmadison 
I added a couple new slides to the presentation on growth. Hopefully they are a reasonable start at addressing Sabrina's point today about the Kline data set.  
https://docs.google.com/presentation/d/1tJ0B4Ch14V3xKqTK0jz_4sOj8VQli1NA7fQ38SRBRHw/edit?usp=sharing

I think the Butler data set is preferred for a multitude of reasons (it's sex-specific, uses multiple age readers instead of one, and has a much larger sample size). Do we collectively agree on that point? If so, do we agree to move forward with the Butler Schnute lognormal curves for the base 2023 model, or do we feel additional analysis is warranted at this time?

In terms of moving forward, I think we can simply acknowledge in the text that the Kline data set exists, but we chose not to use it for the aforementioned reasons. If we want to go further than that, we can increase the range of the sensitivity analysis to include the Kline data. Do you all have thoughts/preferences?

Note that I emailed DK on 2/24/23 on the topic of the different growth curves to get her opinion and will document her response here.

**Reply by AndreaNOdell** [reply]
I agree with you that the Butler data is the preferred dataset for the reasons you mentioned above and agree to move forward with that dataset and the Schnute Lognormal growth model with +/- 10% on both the L1 and L2 parameters for the base 2023 model. We'll see what Donna says, but I think this is a pretty solid direction for now.

**Reply by JaneSullivan-NOAA** [reply]
@AndreaNOdell thank you!

I emailed the following to DK last week:

> We noticed when analyzing the two data sets that fish from your study appeared to grow faster and reach larger sizes than fish from the Butler study. We are unsure if these differences should be attributed to ageing error/bias or spatial variability in growth or size-at-age. I am curious if you and John ever discussed this issue, and if you had any insight into what might be driving these differences in growth. 

Her amazing response:
> The answer to your question is absolutely! This is why John was so enthusiastic about the radiochemistry. He had early reservations about the ageing because the otoliths are so messy! Have you seen the sections? He and his tech had a very tough time reaching the same ages in multiple reads. I don’t think they ever came up with systematic rules for counting annuli. The counted everything they saw using transmitted light. These two species grow too slowly and have pressure-dependent proteins that make them impossible to raise or use mark recapture techniques for validation. Gil vanDykehousen from the Monterey Bay Aquarium managed to keep a couple of Shortspines captured on a seamount alive for about 5 years. Even with aquarium feeding regimes, they grew very little.
>  Regarding the data, there is definitely a systematic difference between our two datasets as evidenced by your graph. Mine are younger at size. However, when we conducted the comparative study (reader 1 me, 2 Mary Yoklavich, 3 John Butler), there were relatively wide errors at older ages, but nothing systematic emerged. This could have indicated some adjustment in personal criteria since we had no written rules as yet. My ageing background at that time had been with dailies, which are also narrow and somewhat irregular. So, since I was having a lot of trouble with consistency, I fell back on that training and used reflected light which has a tendency to obscure really faint marks. I also took good measurements from the newly settled group and made that my 2 yr start point if the core was messy. I also at least considered the maturity check when counting to stay on track. Those criteria may have influenced my younger ages. Reality is likely somewhere in the middle between John’s ages and mine though there is likely some geographic variability that may be influencing the differences as well. My specimens were all collected from a very narrow latitudinal range. This illustrates the need to establish specific criteria as soon as possible in a species ageing program - which you all do now. Back then, that process was just gaining traction.
>  Because we were having so much difficulty coming to a consensus on ages, we decided to use radiochemistry based on otolith material weight as a proxy for age just to verify whether we were in the right ballpark. Those samples were mostly consistent with the ones based on age though with the errors we saw, they would have had to be pretty far off for significance.
>  All of these data and techniques must be viewed through the lens of toleration for errors and uncertainty. You’ll see in my thesis that the radiochemistry graphs have error bars both vertically (radiochemistry) and horizontally (ageing). 
>  I have a couple of photos of sections if you’d like to see them. You really have to see them next to some of the rockfish sections to really appreciate the challenge in ageing these fish.

Based specifically on her comment that "Reality is likely somewhere in the middle between John’s ages and mine though there is likely some geographic variability that may be influencing the differences as well", perhaps it is worthwhile to increase to upper bound of the sensitivity curve. If the upper bound is defined as a 25% increase of Lengths at A1 and A2, and the lower bound is defined as a 10% increase of Lengths at A1 and A2, we get the following results:
![image](https://user-images.githubusercontent.com/68303089/221680612-d6446c07-3cd9-4d96-8d44-f61edf3317fa.png)
This range appears to adequately include the Kline data.

**Reply by AndreaNOdell** [reply]
wow, I so admire her sharp memory! @JaneSullivan-NOAA this looks great. I'm glad we can utilize at least some of the information from DK's dataset. Thank you for the update!

**Reply by sabrinabeyer** [reply]
Yes, looks good! Thanks @JaneSullivan-NOAA for emailing Donna those questions. Her insights are helpful! I like the idea of the 25% sensitivity to encompass the uncertainty in growth and the different findings between the two studies.

**Reply by JaneSullivan-NOAA** [reply]
Awesome, thank you for your work on this issue! I agree its best that the sensitivity include the differences between the two studies.

**Comment by JaneSullivan-NOAA** [comment]
Option for archiving: https://www.fisheries.noaa.gov/inport/

**Comment by Ovec8hkin** [comment]
@sabrinabeyer @JaneSullivan-NOAA have you seen [this package](https://github.com/pfmc-assessments/nwfscAgeingError) by pfmc-assessments for calculating aging error? Since there are some concerns about the aging accuracy of the Butler data, this might be something worth looking at.

**Reply by JaneSullivan-NOAA** [reply]
@Ovec8hkin thanks for sharing! I have seen this package. An ageing error matrix would only be used if we were fitting age composition data.

**Reply by sabrinabeyer** [reply]
Yes, I've seen it and I think it feeds into an option in SS3 to input age-specific error estimates. But agreed that we're not fitting to age composition data.

**Comment by sabrinabeyer** [comment]
As suggested by @JaneSullivan-NOAA, I am moving this comment thread to the discussion page for the growth team. @shipmadison and @AndreaNOdell @MatthVeron 

I think there is a good plan in place to deal with the uncertainty in the growth curve. I hope running a sensitivity test at +/- 25% will give us insight into if the uncertainty in growth is a big deal or not in the model.

I wanted to note that although there is a fairly large precision error in reading the growth increments on the otoliths, the Butler et al 1995, Kline 1996, and Calliet et al 2020 studies found that these fish are likely long-lived through a radiometric analysis. See Figure 3b in Calliet et al 2020 showing the increment age estimate by otolith readers and radiometric estimate with error. This figure uses the Kline 1996 data.

My (non-expert) understanding is that radiometric age validation is an indirect validation method to confirm if fish are long-lived or not, but not a direct validation that each of the growth increments truly represents one year of growth.

I haven't looked at the other studies, but I think the radiometric validation for longevity is still an acceptable method. Although I think there are quite a few assumptions of the method, which may not always be met. 

Anyway, this conversation is fairly tangential but I wanted to point out that there is still evidence of longevity in this species. Unless there are contradictions to the Calliet study that others have found! I would love to know more about these methods of age validation!

**Comment by MatthVeron** [comment]
Hey guys, That is something I forgot to share with you all but thanks Ian, here we are: the link towards the [CAPAM workshop](http://www.capamresearch.org/growth/workshop) he just talked about. 
It can be useful to see how we can model growth in SS and better understand all the options. This [link](https://github.com/iantaylor-NOAA/SSgrowth) can particularly be useful as there are some examples about growth models in SS.

**Comment by JaneSullivan-NOAA** [comment]
Notes from during the data workshop:

> Just to provide some additional background on the ageing issue, Patrick McDonald, who leads our ageing lab, undertook to test whether he could age shortspines reliably in the last 2-4 years. He conducted double reads on structures that were randomized between reads, and he was not satisfied with his ability to produce multiple ages for the same structure that had acceptable rates of variability.

> I wanted people to be aware that we tried to see if we could age shortspine otoliths in the traditional manner when we thought it was likely that the stock would be assessed again before too long, and were not able to. There may be some other possible approaches that could be explored in the future, but those are pretty speculative at this point.
